airbyte-workers/src/main/java/io/airbyte/workers/RecordSchemaValidator.java

```diff
@@ -49,7 +49,7 @@ public void validateSchema(final AirbyteRecordMessage message) throws RecordSche
     try {
       validator.ensure(matchingSchema, messageData);
     } catch (final JsonValidationException e) {
-      throw new RecordSchemaValidationException(String.format("Record schema validation failed. Errors: %s", e.getMessage()), e);
+      throw new RecordSchemaValidationException(String.format("Record schema validation failed for %s. Expected schema: %s", messageStream, matchingSchema));
```

Comment:
Contrary to what I thought earlier, the json schema validator actually does log the data from the invalid record, so I'm changing this to just log the stream name & expected schema.

Reply:
Are you sure we are actually logging the record data? Looking at the examples in the linked issue, and looking at the implementation of the `ensure` json schema validator method, it seems like it should only be logging the name of the field that had a wrong type, and the expected schema: https://github.com/airbytehq/airbyte/blob/91d6d29085b5466a56277981996ea7087b8256d7/airbyte-json-validation/src/main/java/io/airbyte/validation/json/JsonSchemaValidator.java#L62-L72

Reply:
yeah, I thought so too looking at the example issues, but when I tested locally I saw errors that showed the actual data. objectJson in the `ensure` method does have the record data

Reply:
here's an example error message I'm seeing locally:
```
Errors: $.properties.createdate: 2021-02-23 20:03:18 is an invalid date-time, $.properties.hs_lastmodifieddate: 2021-02-23 20:03:18 is an invalid date-time 
```

Reply:
Ah okay, we should change that then. If there is some way we could preserve the part of the validation error that indicates what type of validation error occurred, e.g.
```
Errors: $.committer: null found, object expected, $.author: null found, object expected 
```
that would be desired since that would help us and our users understand how to fix the schema. If we can't get that, then I think what you have here should be okay

Reply:
I looked into this and it looks like it's happening within the json schema validator library, so I'm not sure it's possible